enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,510 @@
+}
+
+type VCenter struct {
+  // VCenter is the domain name or IP address of the vCenter.
+  // +required
+  Server string `json:"server,omitempty"`
+
+  // Port is the TCP Port of the vCenter endpoint
+  // +required
+  Port uint `json:"port,omitempty"`
+
+  // User is the username that will be used to connect to vCenter
+  // +required
+  User string `json:"user,omitempty"`
+
+  // Password is the password for the user to use to connect to the vCenter.
+  // +required
+  Password string `json:"password,omitempty"`
+
+  // +required
+  Regions []Region `json:"regions,omitempty"`
+}
+
+type Region struct {
+  // The name is the region name to be created as a tag
+  // in vSphere and attached to a datacenter and used
+  // with the cloud provider config
+  // +required
+  Name string `json:"name,omitempty"`
+
+  // Datacenter is the name of the datacenter to use in the vCenter.
+  // +required
+  Datacenter string `json:"datacenter,omitempty"`
+
+  // +required
+  Zones []Zone `json:"zones,omitempty"`
+}
+
+type Zone struct {
+  // Name is the zone name to be created as a tag in vSphere
+  // and attached to a cluster and used with the cloud provider config
+  // +required
+  Name string `json:"name,omitempty"`
+  // +required
+  Cluster string `json:"cluster,omitempty"`
+  // Network specifies the name of the network to be used by the cluster.
+  // +required
+  Network string `json:"network,omitempty"`
+  // ResourcePool is the absolute path of the resource pool where virtual machines will be
+  // created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+  // +optional
+  ResourcePool string `json:"resourcepool,omitempty"`
+  // Datastore is the default datastore to use for provisioning volumes.
+  // +required
+  Datastore string `json:"datastore,omitempty"`
+}
+```
+
+Add [platform validation](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/platform.go)
+for the new struct fields that are required.
+
+#### MachinePool
+
+The `MachinePool` [struct](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/machinepool.go#L5-L26)
+needs a single change to include the zones.
+The `Zones` like the other cloud providers will determine the location
+where the virtual machine will reside within a vSphere environment.
+
+```golang
+type MachinePool struct {
+...
+  Zones []string `json:"zones,omitempty"`
+}
+```
+
+Add [machinepool validation](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/machinepool.go)
+the zone name has a corresponding entry in `Regions.Zones`.
+
+#### Cloud Config
+
+The out-of-tree CCM is required for this change. The out-of-tree CCM also updates
+the cloud-config ini configuration. Below is an example of what the
+cloud-config will change to. A `VirtualCenter` section will be added per
+vcenter in `vcenters`. `datacenters` is a comma-delimited string that will contain
+all the datacenters per region.
```

Comment:
In a world where this enhancement is implemented, only the CSI driver and CCM will be consuming the zonal information right? 

The OpenStack folks are working on a config transformer layer for CCM right now. Perhaps rather than injecting this into existing cloud configs, we should just use the vSphere config transformer to inject this information to the CCM where it is needed.